buildkite/bazel_bench.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This script is downloaded and executed by BuildKite when the pipeline starts.
+Runs bazel-bench on the defined projects, on every platforms the project runs
+on.
+"""
+
+import bazelci
+import datetime
+import shutil
+import subprocess
+import tempfile
+import yaml
+
+
+def _platform_path_str(posix_path):
+  """Converts the path to the appropriate format for platform."""
+  if os.name == "nt":
+    return posix_path.replace("/", "\\")
+  return posix_path
+
+
+# TODO(leba): Make these configurable via flags to the script.
+# TMP has different values, depending on the platform.
+TMP = tempfile.gettempdir()
+# The path to the directory that stores the bazel binaries.
+BAZEL_BINARY_BASE_PATH = _platform_path_str("%s/.bazel-bench/bazel-bin/" % TMP)
+PROJECTS = [
+    {
+        "name": "Bazel",
+        "git_repository": "https://github.com/bazelbuild/bazel.git",
+        "bazel_command": "build ..."
+    }
+]
+BAZEL_REPOSITORY = "https://github.com/bazelbuild/bazel.git"
+DATA_DIRECTORY = _platform_path_str("%s/.bazel-bench/out/" % TMP)
+RUNS = 3
+BIGQUERY_TABLE = "bazel_playground:bazel_bench:europe-west2"
+
+
+def get_bazel_commits(day):
+  """Get the commits from a particular day.
+
+  Get the commits from 00:00 of day to 00:00 of day + 1.
+
+  Args:
+    day: a datetime.date the day to get commits.
+
+  Return:
+    A list of string (commit hashes).
+  """
+  day_plus_one = day + datetime.timedelta(days=1)
+  args = [
+      "git",
+      "log",
+      "--pretty=format:'%H'",
+      "--after='%s'" % day.strftime("%Y-%m-%d 00:00"),
+      "--until='%s'" % day_plus_one.strftime("%Y-%m-%d 00:00"),
+      "--reverse"
+  ]
+  command = subprocess.Popen(args, shell=True, stdoud=subprocess.PIPE)
+  return [line.strip() for line in command.stdout]
+
+
+def get_platforms(project_name):
+    """Get the platforms on which this project is run on BazelCI.
+
+    Args:
+      project_name: a string: the name of the project. e.g. "Bazel".
+
+    Returns:
+      A list of string: the platforms for this project.
+    """
+    http_config = bazelci.DOWNSTREAM_PROJECTS[project_name]["http_config"]
+    configs = bazelci.fetch_configs(http_config, None)
+    tasks = configs["tasks"]
```

Comment:
some configs still use the legacy field "platforms":

tasks = configs.get("tasks") or configs.get("platforms")

Reply:
`fetch_configs` already accounts for the legacy format:
https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/bazelci.py#L573

Reply:
Yeah, I forgot that I had added that code :D